### PR TITLE
Add service and controller for customer

### PR DESCRIPTION
**Pull request by muhikira**

### CODE REVIEW

File: src/main/java/com/codemaniac/flightreservation/entity/Customer.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @NotNull
+    Date dateOfBirth;
+    @NotNull
+    public String nationality;
+
+    @NotNull
+    @Size(min=7, message="Passport should have atleast 2 characters")
+    public String passportNumber;
+    @NotNull
+
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/entity/Customer.java:45]: remove unnecessary space here

File: src/main/java/com/codemaniac/flightreservation/controller/CustomerController.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @GetMapping
+    public ResponseEntity<List<Customer>> getAllCustomers() {
+        List<Customer> customers = customerService.getAllCustomers ();
+        return new ResponseEntity<> (customers, HttpStatus.OK);
+    }
+
+    @PutMapping("{id}")
+
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/controller/CustomerController.java:39]: remove unnecessary space

File: src/main/java/com/codemaniac/flightreservation/repository/CustomerRepository.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.codemaniac.flightreservation.entity.Customer;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Map;
+
+@Repository
+public interface CustomerRepository extends JpaRepository <Customer, Long> {
+//    Map<Object, Object> findAllById(Long customerId);
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/repository/CustomerRepository.java:11]: remove commented/unused code

File: src/main/java/com/codemaniac/flightreservation/service/CustomerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    public Customer createrCustomer(Customer customer) {
+
+        return customerRepository.save (customer);
+    }
+
+    @Override
+    public Customer getCustomerById(Long customerId) {
+        Optional<Customer> optionalCustomer = customerRepository.findById (customerId);
+
+        return optionalCustomer.get ();
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/service/CustomerServiceImpl.java:22]: remove space get and brackets

File: src/main/java/com/codemaniac/flightreservation/service/CustomerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        existingCustomer.setEmail (customer.getEmail ());
+        existingCustomer.setNationality (customer.getNationality ());
+        existingCustomer.setFirstName (customer.getFirstName ());
+        existingCustomer.setOtherName (customer.getOtherName ());
+        existingCustomer.setLastName (customer.getLastName ());
+        existingCustomer.setPassportNumber (customer.getPassportNumber ());
+        existingCustomer.setFrequentFlyerNumber (customer.getFrequentFlyerNumber ());
+        existingCustomer.setPhoneNumber (customer.getPhoneNumber ());
+        Customer updatedCustomer = customerRepository.save(existingCustomer);
+
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/service/CustomerServiceImpl.java]: remove whitespace between line 48 and 50

File: src/main/java/com/codemaniac/flightreservation/service/CustomerService.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+
+import java.util.List;
+
+public interface CustomerService {
+    Customer createrCustomer( Customer customer);
+    Customer getCustomerById(Long customerId);
+
+    List<Customer> getAllCustomers();
+    Customer updateCustomer (Customer customer);
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/service/CustomerService.java:13]: update customer method should also have a customer Id as a parameter. Remember you must first find the customer by id before you update changes using the customer object from the request body

File: src/main/java/com/codemaniac/flightreservation/controller/CustomerController.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    public ResponseEntity<Customer> updatedCustomer(@PathVariable("id") Long id,
+                                                  @RequestBody Customer customer) {
+        customer.setId  (id);
```
Comment (rmaheto) [inline src/main/java/com/codemaniac/flightreservation/controller/CustomerController.java]: I would not do this.  Before a customer is updated, the customer should first of all exist in the database. which means you would first of all search for that customer in the customer service and then only update the found customer with the values from the customer request body just as I commented in the  update customer method in the service interface

File: pom.xml [hunk 8]
```diff
@@ -52,6 +52,34 @@
...
+            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+            <version>2.0.4</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
+            <version>7.0.1.Final</version>
+        </dependency>
+        <dependency>
```
Comment (rmaheto) [inline pom.xml:68]: do we still need this even though we have the javax.validation? I am not sure about this could you please find out?

File: pom.xml
Comment (rmaheto) [inline pom.xml:1]: test

Comment (muhikira) [inline pom.xml:1]: test

**Review by rmaheto — APPROVED** [review]

**Review by rmaheto — APPROVED** [review]

**Pull request merged**